.travis.yml

```diff
@@ -29,13 +29,27 @@ jobs:
       install:
         - python3 codalab_service.py build services --version ${TRAVIS_BRANCH} --pull $([ -z "${CODALAB_DOCKER_USERNAME}" ] || [ "$TRAVIS_PULL_REQUEST_BRANCH" != "" ] || echo "--push")
         - python3 codalab_service.py start --services default monitor --version ${TRAVIS_BRANCH}
+        - pip install -r requirements.txt
+      before_script:
+        # For testing CodaLab in Chrome
+        - wget https://chromedriver.storage.googleapis.com/2.41/chromedriver_linux64.zip
+        - unzip chromedriver_linux64.zip
+        - sudo mv chromedriver /usr/bin/chromedriver
+        - sudo chown root:root /usr/bin/chromedriver
+        - sudo chmod +x /usr/bin/chromedriver
+        # For testing CodaLab in Firefox
+        - wget https://github.com/mozilla/geckodriver/releases/download/v0.21.0/geckodriver-v0.21.0-linux64.tar.gz
+        - mkdir geckodriver
+        - tar -xzf geckodriver-v0.21.0-linux64.tar.gz -C geckodriver
+        - export PATH=$PATH:$PWD/geckodriver
       script:
         - python3 codalab_service.py test --version ${TRAVIS_BRANCH} default
           # Make sure restarting worker doesn't cause any issues (ie in serialization/deserialization)
         - docker restart codalab_worker_1 && python3 codalab_service.py test run --version ${TRAVIS_BRANCH}
         - docker stop codalab_worker_1
         - python3 codalab_service.py start --services worker --version ${TRAVIS_BRANCH} --shared-file-system
         - python3 codalab_service.py test --version ${TRAVIS_BRANCH} --shared-file-system basic run read write kill resources netcat netcurl workers
+        - python3 tests/ui/ui_tester.py --headless
```

Comment:
How long do these tests run? If it takes very long, it may be worth running the ui tests in a parallel job to running the `test_cli.py` tests.

Reply:
It runs for about ~20 seconds.